src/main/java/jnr/enxio/channels/Common.java

```diff
@@ -92,13 +98,18 @@ long read(ByteBuffer[] dsts, int offset, int length)
 
     int write(ByteBuffer src) throws IOException {
 
-        ByteBuffer buffer = ByteBuffer.allocate(src.remaining());
+        int n;
+        if (!src.hasArray() && !src.isDirect()) {
+            ByteBuffer buffer = ByteBuffer.allocate(src.remaining());
 
-        buffer.put(src);
+            buffer.put(src);
```

Comment:
this is incorrect as per #50 as the src position is updated to limit, which it should be updated by n (bytes written)